src/runtime/native.rs

```diff
@@ -265,6 +265,17 @@ impl JsRuntimeApi for JsRuntime {
             script_fn.call(scope, output.into(), &[]);
         });
     }
+
+    fn frame_end(&self, world: &mut World) {
+        let this: &mut JsRuntimeInner = &mut self.borrow_mut();
+
+        // Run the JS event loop
+        with_world(world, &mut this.runtime, |runtime| {
+            if let Err(e) = pollster::block_on(runtime.run_event_loop(true)) {
```

Comment:
> Since running the event loop doesn't depend on the `World`...

It actually does depend on world right now because the garbage collection hook needs access to the `JsValueRefs` resource. The world was just a convenient place to store the state, but we could move it.

I'm not sure if moving it helps, though, because, like you said we need mutable access to the runtime, and the runtime might be needed at any Bevy stage, because there might be scripts running at any stage.

As it stands, we don't have any async ops or dynamic imports, so I _think_ that `run_event_loop` should pretty much always finish with nothing but synchronous tasks, and we probably don't need to worry about it unless we expose any async ops. And we may not need scripts to do anything async anyway, since they're currently expected to run synchronous like a Bevy system.